src/box/txn_limbo.c

```diff
@@ -902,6 +915,10 @@ txn_limbo_read_confirm(struct txn_limbo *limbo, int64_t lsn)
 		}
 		txn_limbo_complete_success(limbo, e);
 	}
+	if (limbo->confirmed_lsn < lsn) {
```

Comment:
Nit: in the "[limbo: collect the checkpoint on last txn's commit](https://github.com/tarantool/tarantool/pull/11605/commits/7be67e4149bb35e7e9a53bd45bdeb077066bcbd7)" you move it up, in "limbo: make an illusion of CONFIRMs not batching " you move it back, down, but without the comment